src/Features/Core/Portable/ExtractClass/ExtractClassWithDialogCodeAction.cs

```diff
@@ -0,0 +1,260 @@
+                    updatedDocument.Project.Solution,
+                    symbolMapping,
+                    newType,
+                    extractClassOptions.MemberAnalysisResults,
+                    cancellationToken).ConfigureAwait(false);
+
+                // After all the changes, make sure we're using the most up to date symbol 
+                // as the destination for pulling members into
+                var documentWithTypeDeclaration = solutionWithUpdatedOriginalType.GetRequiredDocument(updatedDocument.Id);
+                newType = await GetNewTypeSymbolAsync(documentWithTypeDeclaration, newType, cancellationToken).ConfigureAwait(false);
+
+                // Use Members Puller to move the members to the new symbol
+                var finalSolution = await PullMembersUpAsync(
+                    solutionWithUpdatedOriginalType,
+                    newType,
+                    symbolMapping,
+                    extractClassOptions.MemberAnalysisResults,
+                    cancellationToken).ConfigureAwait(false);
+
+                return new[] { new ApplyChangesOperation(finalSolution) };
+            }
+            else
+            {
+                // If user click cancel button, options will be null and hit this branch
+                return SpecializedCollections.EmptyEnumerable<CodeActionOperation>();
+            }
+        }
+
+        private async Task<Solution> PullMembersUpAsync(
+            Solution solution,
+            INamedTypeSymbol newType,
+            AnnotatedSymbolMapping symbolMapping,
+            ImmutableArray<ExtractClassMemberAnalysisResult> memberAnalysisResults,
+            CancellationToken cancellationToken)
+        {
+            using var _ = ArrayBuilder<(ISymbol member, bool makeAbstract)>.GetInstance(out var pullMembersBuilder);
+
+            var remainingResults = new List<ExtractClassMemberAnalysisResult>(memberAnalysisResults);
+
+            foreach (var documentId in symbolMapping.DocumentIds)
+            {
+                if (remainingResults.Count == 0)
+                {
+                    // All symbols have been taken care of
+                    break;
+                }
+
+                var document = solution.GetRequiredDocument(documentId);
+
+                var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+                var typeDeclaration = root.GetAnnotatedNodes(symbolMapping.TypeNodeAnnotation).SingleOrDefault();
+                if (typeDeclaration == null)
+                {
+                    continue;
+                }
+
+                foreach (var memberAnalysis in remainingResults.ToArray())
```

Comment:
instead of .toarray to operate on a copy, just have a single loop to get the items of interest, and another loop to then remove from remainingResult.